controllers/mypage-controllers.js

```diff
@@ -0,0 +1,75 @@
+import bcrypt from 'bcrypt';
+import prisma from '../lib/prismaclient.js';
+
+export async function userInfo(req, res) {
+  const userId = req.user.id;
+
+  const user = await prisma.user.findUnique({
+    where: { id: userId },
+  });
+
+  if (!user) return res.status(401).json({ message: 'Unauthorized' });
+
+  const { password: _, ...userInfo } = user;
+  res.status(200).json(userInfo);
+}
+
+export async function updateUserInfo(req, res) {
+  const userId = req.user.id;
+  const { email, nickname, image, ...secret } = req.body;
+  // 변경 가능한 정보를 제외 한 나머지 정보는 secret에 담아둠
+
+  if (email) return res.status(400).json({ message: 'Cannot updete email' });
+
+  // user 검증
+  const user = await prisma.user.findUnique({ where: { id: userId } });
+
+  if (!user) return res.status(401).json({ message: 'Unauthorized' });
```

Comment:
요 두개의 위치가 서로 바뀌어야 할 것 같아요!

```suggestion
  // user 검증
  if (!user) return res.status(401).json({ message: 'Unauthorized' });
  const user = await prisma.user.findUnique({ where: { id: userId } });
```

Reply:
25번 const user.. 구문이 토큰으로 전달 받은 userId를 가지고 서버에 존재하는 유저인지 확인 하는 구문인데,
만약 두 줄이 바뀌게 되면 if문 안에 있는 !user에 false 값이 들어가 에러 메세지를 출력 하게 될 것 같습니다.  

전체 파일을 확인 해 보니, 이전에 선언한게 없는데 어떤 이유로 변경 되어야 하는지 궁금합니다^^
(해당 구문을 다른 컨트롤러에도 많이 사용해서, 멘토님의 조언을 듣고 다른 파일들도 같이 변경하려고요)

Reply:
아앗! 제가 코드를 잘못 읽은 것 같네요, `!user` 를 `!userId` 로 읽었습니다 ㅎㅎ
그렇다면 지금 그대로 두시는게 나을 것 같아요!